packages/core/base/src/services/useCrossmintEvents.ts

```diff
@@ -1,23 +1,62 @@
 import { CheckoutEvents, ListenToMintingEventsProps, ListenerType } from "../models/events";
-import { fetchOrderStatus } from "../utils/payments";
+import { CrossmintCheckoutEvent } from "../models/paymentElement";
+import { getEnvironmentBaseUrl } from "../utils";
 
-export default function useCrossmintEvents() {
+export function useCrossmintEvents({ environment }: { environment?: string } = {}) {
     function listenToMintingEvents({ orderIdentifier }: ListenToMintingEventsProps, cb: ListenerType) {
         const emittedEvents: CheckoutEvents[] = [];
+        const succeededTransactionIdentifiers: string[] = [];
+        const failedTransactionIdentifiers: string[] = [];
 
         const timer = setInterval(async () => {
-            const event = await fetchOrderStatus({ orderIdentifier });
-            if (emittedEvents.includes(event.type)) {
-                return;
-            }
-            cb(event);
-            emittedEvents.push(event.type);
-            if (event.type === CheckoutEvents.MINTING_COMPLETED) {
-                clearInterval(timer);
+            const events = await fetchOrderStatus({ orderIdentifier });
+
+            for (const event of events) {
+                if (emittedEvents.includes(event.type)) {
```

Comment:
this feels a bit messy

can we break this out a bit, maybe like:

if hasSeenEventAlready => handleDuplicateEvent
else => handleNewEvent

Reply:
makes sense. Will improve it today